### ISSUE TITLE
0.3.4: container_definitions is invalid: Error decoding JSON: json: cannot unmarshal bool into Go struct field KeyValuePair.Environment.Value of type string

### ISSUE THREAD

**Issue by alex-quach-arc-boats** [airplanedev/terraform-aws-airplane-agents#20]
```
│ Error: ECS Task Definition container_definitions is invalid: Error decoding JSON: json: cannot unmarshal bool into Go struct field KeyValuePair.Environment.Value of type string
│ 
│   with module.airplane_agent.aws_ecs_task_definition.agent_task_def,
│   on .terraform/modules/airplane_agent/main.tf line 179, in resource "aws_ecs_task_definition" "agent_task_def":
│  179:   container_definitions = jsonencode([
│  180:     {
│  181:       name  = "airplane-agent"
│  182:       image = "us-docker.pkg.dev/airplane-prod/public/agentv2:1"
│  183:       environment = [
│  184:         { name = "AP_API_HOST", value = var.api_host },
│  185:         { name = "AP_API_TOKEN", value = var.api_token },
│  186:         { name = "AP_API_TOKEN_SECRET_ARN", value = var.api_token_secret_arn },
│  187:         { name = "AP_AUTO_UPGRADE", value = "true" },
│  188:         { name = "AP_AGENT_IMAGE", value = "us-docker.pkg.dev/airplane-prod/public/agentv2:1" },
│  189:         { name = "AP_DEBUG_LOGGING", value = var.debug_logging },
│  190:         { name = "AP_DEFAULT_CPU", value = var.default_task_cpu },
│  191:         { name = "AP_DEFAULT_MEMORY", value = var.default_task_memory },
│  192:         { name = "AP_DRIVER", value = "ecs" },
│  193:         { name = "AP_ECS_CLUSTER", value = var.cluster_arn == "" ? aws_ecs_cluster.cluster[0].arn : var.cluster_arn },
│  194:         { name = "AP_ECS_EXECUTION_ROLE", value = aws_iam_role.default_run_role.arn },
│  195:         { name = "AP_ECS_LOG_GROUP", value = aws_cloudwatch_log_group.run_log_group.name },
│  196:         { name = "AP_ECS_REGION", value = data.aws_region.current.name },
│  197:         { name = "AP_ECS_SECURITY_GROUPS", value = length(var.vpc_security_group_ids) > 0 ? join(",", var.vpc_security_group_ids) : join(",", [for sg in module.agent_security_group : sg.security_group_id]) },
│  198:         { name = "AP_ECS_SUBNETS", value = join(",", var.subnet_ids) },
│  199:         { name = "AP_ENV_SLUG", value = var.env_slug },
│  200:         { name = "AP_LABELS", value = join(" ", concat(["airplane_installer:terraform_ecs"], [for key, value in var.agent_labels : "${key}:${value}"])) },
│  201:         { name = "AP_TEAM_ID", value = var.team_id },
│  202:         { name = "AP_RUNNER_TEMPORAL_HOST", value = var.temporal_host },
│  203:         { name = "AP_LOCK_KEY", value = "fargate-${random_uuid.lock_key.result}-${var.team_id}" },
│  204:       ]
│  205:       logConfiguration = {
│  206:         logDriver = "awslogs"
│  207:         options = {
│  208:           "awslogs-region"        = data.aws_region.current.name
│  209:           "awslogs-group"         = aws_cloudwatch_log_group.agent_log_group.name
│  210:           "awslogs-stream-prefix" = "agentv2"
│  211:         }
│  212:       }
│  213:     }
│  214:   ])
```

```
Terraform v1.3.0
on linux_amd64
+ provider registry.terraform.io/hashicorp/archive v2.2.0
+ provider registry.terraform.io/hashicorp/aws v4.9.0
+ provider registry.terraform.io/hashicorp/dns v3.2.3
+ provider registry.terraform.io/hashicorp/local v2.2.2
+ provider registry.terraform.io/hashicorp/null v3.1.1
+ provider registry.terraform.io/hashicorp/random v3.1.2
+ provider registry.terraform.io/hashicorp/template v2.2.0
```

**Comment by alex-quach-arc-boats** [comment]
Thanks for looking at it!

**Comment by andy-hubble** [comment]
Seeing this as well - downgrading to version "0.3.3" resolved it.

**Comment by alex-quach-arc-boats** [comment]
Looks fixed in 0.3.6! Thanks!
